source/github-helpers/github-token.ts

```diff
@@ -1,5 +1,57 @@
-import {v3} from './api.js';
+import {CachedFunction} from 'webext-storage-cache';
+import * as pageDetect from 'github-url-detection';
+
+// Avoid importing api.js here, there's too much logic/caching we don't need
 import {getToken} from '../options-storage.js';
+import hashString from '../helpers/hash-string.js';
+
+type BaseApiFetchOptions = {
+	apiBase: string;
+	token: string;
+	path: string;
+};
+
+async function baseApiFetch({apiBase, token, path}: BaseApiFetchOptions): Promise<Response> {
+	if (!apiBase.endsWith('/')) {
+		throw new TypeError('apiBase must end with a slash');
+	}
+
+	const response = await fetch(
+		new URL(path, apiBase),
+		{
+			cache: 'no-store',
+			headers: {
+				'User-Agent': 'Refined GitHub',
+				'Accept': 'application/vnd.github.v3+json',
+				'Authorization': `token ${token}`,
+			},
+		},
+	);
+
+	if (!response.ok) {
+		const details = await response.json();
+		throw new Error(details.message);
+	}
+
+	return response;
+}
+
+export const tokenUser = new CachedFunction('token-user', {
```

Comment:
This caches the user as specified in https://github.com/refined-github/refined-github/issues/6446#issuecomment-2346322573 and it'll be used by https://github.com/refined-github/refined-github/issues/7384